applications/serial_lte_modem/src/lwm2m_carrier/slm_at_carrier.c

```diff
@@ -177,7 +177,14 @@ static void on_event_app_data(const lwm2m_carrier_event_t *event)
 {
 	lwm2m_carrier_event_app_data_t *app_data = event->data.app_data;
 
-	rsp_send("\r\n#XCARRIEREVT: %d,%d\r\n", event->type, app_data->buffer_len);
+	char uri_path[sizeof("/65535/65535/65535/65535")]; /* Longest URI possible */
+	uint32_t off = 0;
+
+	for (int i = 0; i < app_data->path_len; i++) {
+		off += snprintf(&uri_path[off], sizeof(uri_path) - off, "/%u", app_data->path[i]);
```

Comment:
+1 on the dynamic sizing and explicitly casting to uint16_t.

Reply:
I'm aware of it, I didn't think it was really needed since the identifiers are defined in the OMA LwM2M specification so the path format will never change. But fair enough, since this is an API interaction, I'm happy to add sanity checks.
* In my opinion, that would only make it more difficult to understand how the size was calculated, whereas 65535 is a direct reference to MAX_ID in the LwM2M spec.
* Sure, I'll use the %hu placeholder.
* They are, I can add a sanity check.

Reply:
> > I'm aware of it, I didn't think it was really needed since the identifiers are defined in the OMA LwM2M specification so the path format will never change.
> 
> And yeah, I am looking at that purely from a code perspective, with the hypothetical possibility that the LwM2M Carrier library code would change with the SLM integration not being updated appropriately. So the suggestions I made are to prevent/mitigate such cases.

I understand. It would be an important regression in the carrier library to break the LwM2M spec, though I also dislike adding complexity and size when there is no real benefit. But as per the suggestion, I will try to ensure that the code is safe by design.

> > In my opinion, that would only make it more difficult to understand how the size was calculated, whereas 65535 is a direct reference to MAX_ID in the LwM2M spec.
> 
> Do you have that MAX_ID defined somewhere in the code? I agree that the hard-coded `6` makes it less obvious. Instead you could do something like `sizeof(STRINGIFY(MAX_ID))`.

We don't, MAX_ID is how the specification refers to it. I can define it here if that's preferred.